bundles/org.openhab.binding.vizio/src/main/java/org/openhab/binding/vizio/internal/handler/VizioHandler.java

```diff
@@ -0,0 +1,572 @@
+        if (metadataRefreshJob == null || metadataRefreshJob.isCancelled()) {
+            this.metadataRefreshJob = scheduler.scheduleWithFixedDelay(this::refreshVizioMetadata, 1, 600,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Update source list (hashes) and other metadata from the Vizio TV
+     */
+    private void refreshVizioMetadata() {
+        synchronized (sequenceLock) {
+            try {
+                InputList inputList = communicator.getSourceInputList();
+
+                List<StateOption> sourceListOptions = new ArrayList<>();
+                inputList.getItems().forEach(source -> {
+                    sourceListOptions.add(new StateOption(source.getName(), source.getValue().getName()));
+                });
+
+                stateDescriptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), SOURCE),
+                        sourceListOptions);
+            } catch (VizioException e) {
+                logger.debug("Unable to retrieve the Vizio TV input list. Exception: {}", e.getMessage(), e);
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+
+        ScheduledFuture<?> metadataRefreshJob = this.metadataRefreshJob;
+        if (metadataRefreshJob != null) {
+            metadataRefreshJob.cancel(true);
+            this.metadataRefreshJob = null;
+        }
+
+        ServiceRegistration<?> localServiceRegistration = serviceRegistration;
+        if (localServiceRegistration != null) {
+            // remove trustmanager service
+            localServiceRegistration.unregister();
+            serviceRegistration = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            logger.debug("Unsupported refresh command: {}", command);
+        } else {
+            switch (channelUID.getId()) {
+                case POWER:
+                    debounce = false;
+                    synchronized (sequenceLock) {
+                        try {
+                            if (command == OnOffType.ON) {
+                                communicator.sendKeyPress(KeyCommand.POWERON.getJson());
+                                powerOn = true;
+                            } else {
+                                communicator.sendKeyPress(KeyCommand.POWEROFF.getJson());
+                                powerOn = false;
+                            }
+                        } catch (VizioException e) {
+                            logger.debug("Unable to send power {} command to the Vizio TV, Exception: {}", command,
+                                    e.getMessage());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
Consider if the message above or even the exception message should be included in this status update to provide more immediate information to the user.